les/client_handler.go

```diff
@@ -99,8 +99,13 @@ func (h *clientHandler) runPeer(version uint, p *p2p.Peer, rw p2p.MsgReadWriter)
 }
 
 func (h *clientHandler) handle(p *peer) error {
-	if h.backend.peers.Len() >= h.backend.config.LightPeers && !p.Peer.Info().Network.Trusted {
-		return p2p.DiscTooManyPeers
+	if !p.Peer.Info().Network.Trusted {
```

Comment:
See my comment regarding adding the incoming peer as a validator peer in the handshake.  If that's done, then this doesn't need to be checked here.